contracts/erc7683/ERC7683Depositor.sol

```diff
@@ -0,0 +1,293 @@
+    uint32 originChainId;
+    /// @dev The timestamp by which the order must be initiated
+    uint32 initiateDeadline;
+    /// @dev The timestamp by which the order must be filled on the destination chain
+    uint32 fillDeadline;
+    /// @dev Arbitrary implementation-specific data
+    /// Can be used to define tokens, amounts, destination chains, fees, settlement parameters,
+    /// or any other order-type specific information
+    bytes orderData;
+}
+
+/// @title ResolvedCrossChainOrder type
+/// @notice An implementation-generic representation of an order
+/// @dev Defines all requirements for filling an order by unbundling the implementation-specific orderData.
+/// @dev Intended to improve integration generalization by allowing fillers to compute the exact input and output information of any order
+struct ResolvedCrossChainOrder {
+    /// @dev The contract address that the order is meant to be settled by.
+    address settlementContract;
+    /// @dev The address of the user who is initiating the swap
+    address swapper;
+    /// @dev Nonce to be used as replay protection for the order
+    uint256 nonce;
+    /// @dev The chainId of the origin chain
+    uint32 originChainId;
+    /// @dev The timestamp by which the order must be initiated
+    uint32 initiateDeadline;
+    /// @dev The timestamp by which the order must be filled on the destination chain(s)
+    uint32 fillDeadline;
+    /// @dev The inputs to be taken from the swapper as part of order initiation
+    Input[] swapperInputs;
+    /// @dev The outputs to be given to the swapper as part of order fulfillment
+    Output[] swapperOutputs;
+    /// @dev The outputs to be given to the filler as part of order settlement
+    Output[] fillerOutputs;
+}
+
+/// @notice Tokens sent by the swapper as inputs to the order
+struct Input {
+    /// @dev The address of the ERC20 token on the origin chain
+    address token;
+    /// @dev The amount of the token to be sent
+    uint256 amount;
+}
+
+/// @notice Tokens that must be receive for a valid order fulfillment
+struct Output {
+    /// @dev The address of the ERC20 token on the destination chain
+    /// @dev address(0) used as a sentinel for the native token
+    address token;
+    /// @dev The amount of the token to be sent
+    uint256 amount;
+    /// @dev The address to receive the output tokens
+    address recipient;
+    /// @dev The destination chain for this output
+    uint32 chainId;
+}
+
+// Data unique to every CrossChainOrder settled on Across
+struct AcrossOrderData {
+    address inputToken;
+    uint256 inputAmount;
+    address outputToken;
+    uint256 outputAmount;
+    uint32 destinationChainId;
+    address recipient;
+    uint32 exclusivityDeadline;
+    address exclusiveRelayer;
+    bytes message;
+}
+
+// Data unique to every attempted order fulfillment
+struct AcrossFillerData {
+    // Filler can choose where they want to be repaid
+    uint32 repaymentChainId;
+}
```

Comment:
Since we can't accurately assess fees if the filler takes repayment elsewhere and we can't enforce this choice on the fill side, thoughts on just resolving _assuming that they will take repayment on origin_ and not having them specify the repayment chain here?

Then the fees and repayment location change if they input a different repayment chain on the destination?